File: src/ros_basics_tutorials/src/talker_listener/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void chatterCallback(const std_msgs::String::ConstPtr& msg){
    ROS_INFO("[Listener] I heard %s\n", msg->data.c_str());
}

int main(int argc, char **argv){
    // initialize a new ROS node
    ros::init(argc, argv, "listener_node");

    // create a subscriber
    // callback function will be executed each time a message is received
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("chatter", 10, chatterCallback);
    
    ros::spin();

    return 0;
} 

File: src/ros_basics_tutorials/src/talker_listener/listener.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def chatter_callback(message):
    rospy.loginfo("I heard %s" % message.data)

def listener():
    # Create and initialize a ROS node
    rospy.init_node('listener', anonymous=True)

    # Create a subscriber object
    rospy.Subscriber('chatter', String, chatter_callback)

    # Start listening
    rospy.spin()

if __name__ == '__main__':
    listener()